src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+
+function validateMuseumPath(selectedPath, showPath) {
+  //input: array of strings that represents roomIDs and specific order a user wishes to go room to room
+  //output: a boolean value indicating whether the path is possible or not
+
+  //edge-case: empty array
+  if (selectedPath.length === 0) return "array cannot be empty";
+
+  //declare a lastRoom variable and set to null originally
+  let lastRoom = null;
+  //make copy of exampleRoomData
+  remainingRooms = exampleRoomData.slice();
+
+  //make sure they start off at coat check room
+  if (selectedPath[0] !== remainingRooms[2].roomId) {
+    return "path must start at 'aIA6tevTne' a.k.a Coat Check Room";
+  } else {
+    lastRoom = exampleRoomData[2].roomId;
+    selectedPath.slice(1);
+  }
+
+  //declare a variable to chop off remaining room options to prevent repeats, and path variable
+  let indexToChopOffRemainingRooms;
+  let gottenPath = "";
+
+  //iterate through path until we are out of bounds
+  for (let i = 0; i < selectedPath.length; i++) {
+    //if next index is out of bounds return true cause we've hit the end of path successfully
+    if (selectedPath[i + 1] === undefined) break;
+    //declare flag variable which will decide if we proceed so long as next room is valid
+    let nextRoomIsValid = false;
+
+    for (let j = 0; j < remainingRooms.length; j++) {
+      if (remainingRooms[j].roomId === lastRoom) {
+        if (remainingRooms[j].connectsTo.includes(selectedPath[i + 1])) {
+          //set next room is valid to true
+          nextRoomIsValid = true;
+          //re-assign index to chop off to current index post-iteration
+          indexToChopOffRemainingRooms = j;
+          lastRoom = selectedPath[i + 1];
+          if (i === selectedPath.length - 2) {
+            gottenPath += `${remainingRooms[j].name} -> `;
+            gottenPath += `${remainingRooms[j + 1].name} -> `;
+            break;
+          } else {
+            gottenPath += `${remainingRooms[j].name} -> `;
+            break;
+          }
+        }
+      }
+    }
+    if (!nextRoomIsValid) return false;
+    remainingRooms.splice(indexToChopOffRemainingRooms, 1);
+  }
+  if (showPath) return gottenPath.slice(0, -3);
+  return true;
+}
+// console.log(validateMuseumPath(valid, true));
+
+/*
+ticketExtraDetails()
+Write a new function called ticketExtraDetails() that takes a string as its argument. If that string matches 
+one of the ticket "extras", provide information about what someone would be able to do based on purchasing 
+that ticket. The format is up to you.
+*/
+
+function ticketExtraDetails(extras) {
+  //edge case - if the extra is not movie education or terrace return error message
+  const validExtras = ["movie", "education", "terrace"];
+
+  if (!Array.isArray(extras) && !validExtras.includes(extras.toLowerCase())) {
+    return "must be a valid extra: 'movie', 'education', or 'terrace'";
+  } else {
+    extras = extras.toLowerCase();
```

Comment:
I believe this will attempt to call `.toLowerCase()` on arrays, which will fail.

One way to manage something that can accept either a single item or a list, is to just assume it's a list and then transform the single item to be so.
```js
if (!Array.isArray(extras)) extras = [extras];
```
Then you can just be solving for a single case.